datanode/sqlstore/ledger.go

```diff
@@ -160,6 +164,85 @@ func (ls *Ledger) Query(
 	return &res, pageInfo, nil
 }
 
+func (ls *Ledger) Export(
+	ctx context.Context,
+	partyID, assetID string,
+	dateRange entities.DateRange,
+) ([]byte, string, error) {
+	if partyID == "" {
+		return nil, "", ErrLedgerEntryExportForParty
+	}
+
+	if assetID == "" {
+		return nil, "", ErrLedgerEntryExportForAsset
+	}
+
+	filter := &entities.LedgerEntryFilter{
+		SenderAccountFilter: entities.AccountFilter{
+			AssetID:  entities.AssetID(assetID),
+			PartyIDs: []entities.PartyID{entities.PartyID(partyID)},
+		},
+	}
+
+	aggregatedEntries, _, err := ls.Query(ctx, filter, dateRange, entities.CursorPagination{})
+	if err != nil {
+		return nil, "", err
+	}
+
+	path := fmt.Sprintf("%s-%s-%s-%s", partyID, assetID, dateRange.Start.String(), dateRange.End.String())
+	file, err := os.Create(path)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to create filename: %s", err.Error())
+	}
+	defer file.Close()
```

Comment:
I'm not sure you would need to write it all to a file.

Reply:
Addresssed - it sends the data as binary payload now
Thanks!

Reply:
At the moment the solution has data encoded to be processed and it can be decoded on the other side as csv.
It can be taken from the response field and dealt with on the FE side.

Do we like this solution or we _specifically_ want to download a file?
Because in that case grpc-gateway will use a custom route for this endpoint? Not that it is a problem, just to be sure - because in the beginning the idea was not to have a file saved in the datanode system.

Reply:
![image (1)](https://user-images.githubusercontent.com/86832315/212416974-71a1970c-61f8-4f96-96df-d70197d27ae4.png)

Reply:
Sorry, I missed this thread previously but I think a direct download makes the most sense:

- it will often be a lot of data, and often the user will want to open it in a 3rd party tool, so it makes sense to me to avoid the overhead of the dapp loading it into memory and decoding it, only to serve the result up as a file for the user to save anyway;
- many users will skip the dapp altogether and just donwload the data with curl or similar.